components/VotePoll.tsx

```diff
@@ -0,0 +1,257 @@
+import React, { Dispatch, SetStateAction } from "react";
+import { useEffect } from "react";
+import { useState } from "react";
+import Button from "@mui/material/Button";
+import Typography from "@mui/material/Typography";
+import { Alert, Snackbar } from "@mui/material";
+import {
+  DragDropContext,
+  Draggable,
+  Droppable,
+  DropResult,
+} from "react-beautiful-dnd";
+import CandidateSelectionBox from "./CandidateSelectionBox";
+import CandidateChoiceBox from "./CandidateChoiceBox";
+import { CreateVote } from "./api";
+import { useFetch } from "../hooks/useFetch";
+
+type Props = {
+  pollData: Poll;
+  setPageAlert: Dispatch<SetStateAction<AlertShape | null>>;
+};
+
+function VotePoll({ pollData, setPageAlert }: Props) {
+  const [_, createVote] = useFetch(CreateVote);
+  const [candidates, setCandidates] = useState<string[]>([]);
+  const [chosenCandidates, setChosenCandidates] = useState<string[]>([]);
+
+  useEffect(() => {
+    if (pollData.candidateList) {
+      setCandidates(pollData.candidateList.map((candidate) => candidate.name));
+    }
+  }, [pollData.candidateList]);
+
+  type DropId = "candidatesDropId" | "chosenCandidatesDropId";
+
+  const dropIdMap = {
+    candidatesDropId: {
+      list: candidates,
+      setList: setCandidates,
+    },
+    chosenCandidatesDropId: {
+      list: chosenCandidates,
+      setList: setChosenCandidates,
+    },
+  };
+
+  const onDragEnd = (result: DropResult) => {
+    const { source, destination } = result;
+    if (!destination) return;
+    const sourceId = source.droppableId as DropId;
+    const destId = destination.droppableId as DropId;
+
+    if (sourceId === destId) {
+      const reOrderedItems = reorder(
+        dropIdMap[sourceId].list,
+        source.index,
+        destination.index
+      );
+
+      dropIdMap[sourceId].setList(reOrderedItems);
+    } else {
+      // Item is being moved from one list to another
+      if (!isValidMove(destId)) {
+        // Create a Toast
```

Comment:
Do you want to create a follow up PR for this?

Reply:
I decided to move the toast into to the valid checking function in case we want to add more invalid cases in the future, forgot to remove this comment.